src/main/java/com/google/cloud/tools/appengine/cloudsdk/CloudSdk.java

```diff
@@ -70,22 +71,25 @@
   @Nullable
   private final File appCommandCredentialFile;
   private final String appCommandOutputFormat;
-  private final WaitingProcessOutputLineListener runDevAppServerWaitListener;
+  private final WaitingProcessOutputLineListener outputLineListener;
+  private final AccumulatingLineListener stdOutput;
 
   private CloudSdk(Path sdkPath,
                    String appCommandMetricsEnvironment,
                    String appCommandMetricsEnvironmentVersion,
                    @Nullable File appCommandCredentialFile,
                    String appCommandOutputFormat,
                    ProcessRunner processRunner,
-                   WaitingProcessOutputLineListener runDevAppServerWaitListener) {
+                   WaitingProcessOutputLineListener outputLineListener,
+                   AccumulatingLineListener stdOutput) {
```

Comment:
They do quite different things. It might make sense to move the AccumulatingLineListener into ProcessRunner itself. It's not exosed in the public API (constructor is private)